app/Http/Controllers/ResumeController.php

```diff
@@ -0,0 +1,192 @@
+                        'status' => 'error',
+                        'message' => 'resume_required',
+                    ],
+                    400
+                );
+            }
+
+            $tempPath = $file->getPathname();
+            $exifReturn = -1;
+            $exifOutput = '';
+            exec('exiftool -json '.escapeshellarg($tempPath), $exifOutput, $exifReturn);
+
+            $exifOutput = json_decode(implode(' ', $exifOutput), true)[0];
+            $fileType = array_key_exists('FileType', $exifOutput) ? $exifOutput['FileType'] : null;
+            $mimeType = array_key_exists('MIMEType', $exifOutput) ? $exifOutput['MIMEType'] : null;
+            $pageCount = array_key_exists('PageCount', $exifOutput) ? $exifOutput['PageCount'] : -1;
+            $exifError = array_key_exists('Error', $exifOutput) ? $exifOutput['Error'] : null;
+
+            $valid = null === $exifError && 'PDF' === $fileType && 'application/pdf' === $mimeType;
+            $pageCountValid = 1 === $pageCount;
+            $exifErrorInvalidType = 'Unknown file type' === $exifError;
+
+            if (! $valid || ! $pageCountValid) {
+                \Log::debug('User resume uploaded for user '.$user->uid.', but was invalid (PDF: '
+                    .($valid ? 'true' : 'false').', one page: '.($pageCountValid ? 'true' : 'false').', Error: '
+                    .$exifError.')');
+                $error = $valid ? 'resume_not_one_page' : 'resume_not_pdf';
+                if ($exifError && ! $exifErrorInvalidType) {
+                    \Log::error('exiftool responded with unknown error');
+                    $error = 'unknown_error';
+                }
+
+                if ($request->has('redirect')) {
+                    return redirect()->route('resume.index', ['resume_error' => $error]);
+                }
+
+                return response()->json(
+                    [
+                        'status' => 'error',
+                        'message' => $error,
+                    ],
+                    400
+                );
+            }
+
+            // Store in the resumes folder with the user's username
+            $file->storeAs('resumes', $user->uid.'.pdf');
+
+            $user->resume_date = now();
+            $user->save();
+
+            if ($request->has('redirect')) {
+                return redirect()->route('resume.index');
+            }
+
+            return response()->json(
+                [
+                    'status' => 'success',
+                ],
+                200
+            );
+        }
+
+        return response()->json(
+            [
+                'status' => 'error',
+                'message' => 'User does not exist or was previously deleted.',
+            ],
+            422
+        );
+    }
+
+    /**
+     * Delete the user's resume.
+     *
+     * @param string $id
+     *
+     * @return \Illuminate\Http\JsonResponse
+     */
+    public function deleteResume(string $id): JsonResponse
```

Comment:
Once the resume book function has been extracted to another controller, this should be renamed to remove `Resume` from the function name.